### PR TITLE
Refactor nullable-related tests, add support for marshalling

### PR DESCRIPTION
**Pull request by sjaensch**
I've refactored the nullable-related tests by @andreashug to better reflect the structure of the other tests: I've split them up into unmarshalling and validation tests and added them the the existing files. I've also added support for marshalling along with some tests as well as a bit of documentation.

### CODE REVIEW

File: tests/unmarshal/unmarshal_primitive_test.py [hunk 1]
```diff
@@ -72,3 +72,15 @@ def test_ref(minimal_swagger_dict):
     special_integer_spec = {'$ref': '#/definitions/SpecialInteger'}
     swagger_spec = Spec.from_dict(minimal_swagger_dict)
     assert 10 == unmarshal_primitive(swagger_spec, special_integer_spec, 10)
+
+
+@pytest.mark.parametrize(['nullable', 'value'],
+                         [(False, 'x'), (True, 'x'), (True, None)])
+def test_nullable(minimal_swagger_spec, value, nullable):
+    """Test scenarios in which validation should pass: (1), (3), (4)"""
```
Comment (laucia) [inline tests/unmarshal/unmarshal_primitive_test.py]: that docstring doesn't means anything now, you need to explain the cases tested explicitely

Comment (sjaensch) [inline tests/unmarshal/unmarshal_primitive_test.py]: Thanks for catching that! The validation is in the other tests now, I removed the docstring.

File: docs/source/models.rst [hunk 2]
```diff
@@ -78,6 +78,25 @@ always disable it if necessary.
     swagger_dict = {..}
     spec = Spec.from_dict(swagger_dict, config={'use_models': False})
 
+Allowing null values for properties
+-----------------------------------
+Typically, bravado-core will complain during validation if it encounters fields with ``null`` values.
+This can be problematic, especially when you're adding Swagger support to pre-existing
+APIs. In that case, declare your model properties as ``x-nullable``:
```
Comment (laucia) [inline docs/source/models.rst]: I was wondering if we should me more explicit about the fact that this is a Vendor Extension and not part of the regular swagger-spec ?

File: tests/unmarshal/unmarshal_object_test.py [hunk 3]
```diff
@@ -236,3 +236,44 @@ def test_recursive_ref_with_depth_n(recursive_swagger_spec):
...
+    assert result == value
+
+
+@pytest.mark.parametrize('nullable', [True, False])
+def test_nullable_no_value(empty_swagger_spec, nullable):
+    content_spec = nullable_spec_factory(False, nullable=nullable)
+    value = {}
+
+    result = unmarshal_object(empty_swagger_spec, content_spec, value)
+    assert result == {'x': None}  # Missing parameters are re-introduced
```
Comment (laucia) [inline tests/unmarshal/unmarshal_object_test.py]: I am probably wrong but I don't get why we would unmarshall to this when nullable is `False` and required is false too. 
That look like the old swaggerpy behavior of adding `None` on not present key.

Comment (sjaensch) [inline tests/unmarshal/unmarshal_object_test.py]: swaggerpy re-added the field with the type's default value, not ``None``. It looks like the current behavior is desired, check out this comment in [unmarshal.py](https://github.com/Yelp/bravado-core/blob/master/bravado_core/unmarshal.py#L124).

File: docs/source/models.rst (hunk 2 above)
Comment (sjaensch) [inline docs/source/models.rst]: Done, what do you think?

**Comment by laucia** [comment]
lg2m 🚢

**Pull request merged**